pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -43,24 +42,24 @@ var (
 )
 
 // IsReadyToReconcile determines if a controller is ready to reconcile or not
-func IsReadyToReconcile(client client.Client, clustercontext *clusterd.Context, namespacedName types.NamespacedName, controllerName string) (cephv1.ClusterSpec, bool, bool, reconcile.Result) {
-	namespacedName.Name = namespacedName.Namespace
-	cephClusterExists := true
+func IsReadyToReconcile(c client.Client, clustercontext *clusterd.Context, namespacedName types.NamespacedName, controllerName string) (cephv1.ClusterSpec, bool, bool, reconcile.Result) {
+	cephClusterExists := false
 
 	// Running ceph commands won't work and the controller will keep re-queuing so I believe it's fine not to check
 	// Make sure a CephCluster exists before doing anything
-	cephCluster := &cephv1.CephCluster{}
-	err := client.Get(context.TODO(), namespacedName, cephCluster)
+	var cephCluster cephv1.CephCluster
+	clusterList := &cephv1.CephClusterList{}
+	err := c.List(context.TODO(), clusterList, client.InNamespace(namespacedName.Namespace))
 	if err != nil {
-		if kerrors.IsNotFound(err) {
-			cephClusterExists = false
-			logger.Errorf("%q: CephCluster resource %q not found in namespace %q", controllerName, namespacedName.Name, namespacedName.Namespace)
-			return cephv1.ClusterSpec{}, false, cephClusterExists, WaitForRequeueIfCephClusterNotReady
-		} else if err != nil {
-			logger.Errorf("%q:failed to fetch CephCluster %v", controllerName, err)
-			return cephv1.ClusterSpec{}, false, cephClusterExists, ImmediateRetryResult
-		}
+		logger.Errorf("%q:failed to fetch CephCluster %v", controllerName, err)
+		return cephv1.ClusterSpec{}, false, cephClusterExists, ImmediateRetryResult
+	}
+	if len(clusterList.Items) == 0 {
+		logger.Errorf("%q: CephCluster resource %q not found in namespace %q", controllerName, namespacedName.Name, namespacedName.Namespace)
```

Comment:
```suggestion
		logger.Errorf("%q: no CephCluster resource found in namespace %q", controllerName, namespacedName.Namespace)
```

`namespacedName.Name` is incorrect so we should probably remove it.

Reply:
I also found a couple other log messages to update